### PR TITLE
First 1.8.9 Forestry Version

### PR DESCRIPTION
**Pull request by Nedelosk**
My first 1.8.9 Forestry version.
Features : 
- All Block Models 
- A test crate block model
  
- Many Item Models
- All BlockStates
- Fix all errors in the code
- JEI Intigreation

Missing : 
- None butterfly item model
- None crate item model

### CODE REVIEW

**Comment by Nedelosk** [comment]
So I'm think the RP is ready.

**Comment by mezz** [comment]
Thanks! I will take a look soon.

**Comment by JavaMan7** [comment]
How do I download 1.8.9 Forestry?

**Comment by mezz** [comment]
@JavaMan7 it is still far from being playable, there is nowhere to download it yet.

**Comment by Ioandar** [comment]
I am so longing for 1.8.9 forestry. Would be epic to see forestry go boldly where no forestry has ever gone before (1.8.9) :)

and a big thank you to all the people putting so much great work in all of forestry!

**Comment by labpaw** [comment]
@mezz legacy chest code should be removed, isn't it?

**Comment by williewillus** [comment]
Your blockstate jsons are all using the incredibly verbose vanilla format that forces you to split everything into separate files.
There's about 4x more JSONs than are necessary in your branch currently (many of the block and itemblock models can all be specified in one blockstate json instead of 1 + 2 \* (number of variants) )
Check out the forge format: https://gist.github.com/RainWarrior/0618131f51b8d37b80a6
And examples of usage: https://github.com/williewillus/Botania/tree/MC18/src/main/resources/assets/botania/blockstates

**Comment by Nedelosk** [comment]
- [ ] Blockstates for Blocks
   - [ ] Apriculture Blocks
   - [ ] Aboriculture Blocks
   - [ ] Core Blocks
   - [ ] Factory Blocks
   - [ ] Mail Blocks
- [ ] Blockstates for Items

**Comment by williewillus** [comment]
You don't have to use them for items, though I would have a "standard_item" parent model so you don't have to copy paste the third person transform into every single model.

**Comment by Nedelosk** [comment]
@williewillus Have you a idea for a new butterfly item model ?

**Comment by williewillus** [comment]
not sure, don't they use the entity to render in 1.7?

**Comment by Nedelosk** [comment]
No in 1.7 they use a IItemRenderer

**Comment by Nedelosk** [comment]
Can I make a json model from a techne model ?

**Comment by williewillus** [comment]
nope

**Comment by williewillus** [comment]
either remake it in json (and use the animation system, optionally) or use a fake TESR, which I wouldn't recommend

File: build.gradle [hunk 1]
```diff
@@ -207,9 +163,8 @@ task apiJar(type: Jar, dependsOn: 'classes') {
 	classifier = 'api'
 }
 
-task sourceJar(type: Jar, dependsOn: 'classes') {
+task srcJar(type: Jar, dependsOn: 'classes') {
```
Comment (liach) [inline build.gradle:166]: `sourceJar {`

File: build.gradle [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-		classpath 'org.ajoberstar:gradle-git:0.8.0'
+		classpath 'net.minecraftforge.gradle:ForgeGradle:2.1-SNAPSHOT'
 	}
 }
 
-apply plugin: 'forge'
-apply plugin: 'maven'
-apply plugin: 'java'
+plugins {
+    id "net.minecraftforge.gradle.forge" version "2.0.2"
```
Comment (liach) [inline build.gradle:19]: You do not need this plugin actually, as you have already used it in buildscript. (`classpath 'net.minecraftforge.gradle:ForgeGradle:2.1-SNAPSHOT'`)

**Comment by Nedelosk** [comment]
@williewillus I get this message http://pastebin.com/9LGCwP62 when i start with this stairs.json
http://pastebin.com/6B8WXV23. What is the problem ?

**Comment by williewillus** [comment]
if you use the forge json format you have to wrap everything inside [ ].

Right now it is thinking "facing=east,half=bottom,shape=inner_right,woodtype=padauk" is the name of a property, and "model" is a value. You need to do "facing=east,half=bottom,shape=inner_right,woodtype=padauk": [{ <stuff> }], <- notice [ ] 
not "facing=east,half=bottom,shape=inner_right,woodtype=padauk": { <stuff> }   

though I feel like there's a better way to do this using the forge json format by breaking up all the subtypes, but haven't experimented with it yet

**Comment by Nedelosk** [comment]
@williewillus Thank-you

**Comment by Nedelosk** [comment]
I have at the moment too little time to work on the block states. Therefore I think @mezz I am ready with this RP. I think another modder can do this, or I do this in the future.

**Comment by mezz** [comment]
Thanks, I will pull this to a branch and begin work soon.

**Comment by Nedelosk** [comment]
Good, in the future I want help you when i have enough time to do this.

**Pull request merged**